src/Simplex/Chat/Controller.hs

```diff
@@ -247,7 +247,7 @@ data ChatCommand
   | ExecChatStoreSQL Text
   | ExecAgentStoreSQL Text
   | SlowSQLQueries
-  | APIGetChats {userId :: UserId, pendingConnections :: Bool}
+  | APIGetChats {userId :: UserId, pendingConnections :: Bool, paginationTs_ :: Maybe ChatPaginationTs, favorite :: Bool, unread :: Bool, search_ :: Maybe String}
```

Comment:
search doesn't take into account filters so this has higher dimensionality than necessary (you can't search **and** filter, but api suggests you can), maybe time to make a type for this:

```haskell
data ChatListQuery
  = CLQNoFilter
  | CLQSearch String
  | CLQFilter ChatListFilters
  -- or for current UI the last one could be simply
  -- | CLQFavoriteOrUnread
  -- and if separate are allowed later we could then add
  -- | CLQFavorite
  -- | CLQUnread

data ChatListFilters = ChatListFilters
  { favorite :: Bool,
    unread :: Bool
  }
```

actually even CLQNoFilter and CLQFilter could be united and if all filters in ChatListFilters are false then it's "no filter"

Reply:
maybe it's a bad idea if we plan to support search while filtering later